NuRadioReco/modules/channelReadoutWindowCutter.py

```diff
@@ -95,39 +95,50 @@ def run(self, event, station, detector):
-            trigger_time_sample = int(np.round(trigger_time_channel * sampling_rate))
-
             channel_id = channel.get_id()
             pre_trigger_time = trigger.get_pre_trigger_time_channel(channel_id)
-            samples_before_trigger = int(pre_trigger_time * sampling_rate)
 
+            pre_trigger_time_channel = trigger_time - pre_trigger_time - channel.get_trace_start_time()
+            
+            # throw warinings or errors depending on if the readout window is outside or too far outside the trace
             trace_length = len(trace)
-            cut_samples_beginning = 0
-            if samples_before_trigger <= trigger_time_sample:
-                cut_samples_beginning = trigger_time_sample - samples_before_trigger
-                if cut_samples_beginning + number_of_samples > trace_length:
-                    logger.warning(("trigger time is sample {} but total trace length is only {} "
-                                    "samples (requested trace length is {} with an offest of {} "
-                                    "before trigger). To achieve desired configuration, trace "
-                                    "will be rolled").format(trigger_time_sample, trace_length,
-                                                             number_of_samples, samples_before_trigger))
-
-                    roll_by = cut_samples_beginning + number_of_samples - trace_length  # roll_by is positive
-                    trace = np.roll(trace, -1 * roll_by)
-                    cut_samples_beginning -= roll_by
-
-            else:
-                roll_by = samples_before_trigger - trigger_time_sample
-                logger.warning(("trigger time is before 'trigger offset window' (requested samples before trigger = {},"
-                                "trigger time sample = {}), the trace needs to be rolled by {} samples first"
-                                " = {}ns").format(samples_before_trigger, trigger_time_sample, roll_by,
-                                                  round(roll_by / sampling_rate/units.ns, 2)))
-
-                trace = np.roll(trace, roll_by)
-
-            # shift trace to be in the correct location for cutting
-            trace = trace[cut_samples_beginning:(number_of_samples + cut_samples_beginning)]
+            if pre_trigger_time_channel < -pre_trigger_time:
+                msg = ("Start of the readout window is more than pre_trigger_time before the start of the "
+                       "trace (trigger time = {}ns, pre-trigger time = {}ns, start of trace {}ns, requested "
+                       "time before trace = {}ns), this would result in rolling over the edge of the trace "
+                       "and is not the intended use of this function").format(
+                            trigger_time, pre_trigger_time, channel.get_trace_start_time(), pre_trigger_time_channel)
+                logger.error(msg)
```

Comment:
one general comment. I prefer f-strings, i.e., `f"trigger time = {trigger_time}"` because they are much easier to read. There is one very good argument to use `.format`: If you for example use `logger.debug` you should use `.format` because the string gets only "produced" if debug is enabled. But in this case where a if condition has to be fulfilled first it is okay to use f-strings

Reply:
just for the future!